src/SymmetricPositiveDefinite.jl

```diff
@@ -0,0 +1,482 @@
+
+cholesky_to_spd(l,w) = (l*l', w*l' + l*w')
+tangent_cholesky_to_tangent_spd!(l,w) = (w .= w*l' + l*w')
+spd_to_cholesky(x,v) = spd_to_cholesky(x,cholesky(x).L,v)
+function spd_to_cholesky(x,l,v)
+    a = l\v
+    w = transpose(l\(a'))
+    # strictly lower triangular plus half diagonal
+    return (l, l*(LowerTriangular(w) - Diagonal(w)/2) )
+end
+
+@doc doc"""
+    distance(M,x,y)
+
+computes the distance on the [`SymmetricPositiveDefinite`](@ref) manifold between `x` and `y`,
+which defaults to the [`LinearAffineMetric`](@ref) induces distance.
+"""
+distance(M::SymmetricPositiveDefinite{N},x,y) where N = distance(MetricManifold(M,LinearAffineMetric()),x,y)
+
+@doc doc"""
+    distance(M,x,y)
+
+computes the distance on the [`SymmetricPositiveDefinite`](@ref) manifold between `x` and `y`,
+as a [`MetricManifold`](@ref) with [`LinearAffineMetric`](@ref). The formula reads
+
+```math
+d_{\mathcal P(n)}(x,y) = \lVert \operatorname{Log}(x^{-\frac{1}{2}}yx^{-\frac{1}{2}})\rVert_{\mathrm{F}}.,
+```
+where $\operatorname{Log}$ denotes the matrix logarithm and $\lVert\cdot\rVert_{\mathrm{F}}$ denotes the
+matrix Frobenius norm.
+"""
+function distance(M::MetricManifold{SymmetricPositiveDefinite{N},LinearAffineMetric},x,y) where N
+    s = real.( eigvals( x,y ) )
+    return any(s .<= eps() ) ? 0 : sqrt(  sum( abs.(log.(s)).^2 )  )
+end
+
+@doc doc"""
+    distance(M,x,y)
+
+computes the distance on the manifold of [`SymmetricPositiveDefinite`](@ref)
+nmatrices, i.e. between two symmetric positive definite matrices `x` and `y`
+with respect to the [`LogCholeskyMetric`](@ref). The formula reads
+
+````math
+d_{\mathcal P(n)}(x,y) = \sqrt{
+ \lVert \lfloor l \rfloor - \lfloor k \rfloor \rVert_{\mathrm{F}}^2
+ + \lVert \log(\operatorname{diag}(l)) - \log(\operatorname{diag}(k))\rVert_{\mathrm{F}}^2 },
+````
+where $l$ and $k$ are the cholesky factors of $x$ and $y$, respectively,
+$\lfloor\cdit\rfloor$ denbotes the lower triangulr matrix of its argument,
+and $\lVert\cdot\rVert_{\mathrm{F}}$ denotes the Frobenius norm.
+"""
+distance(M::MetricManifold{SymmetricPositiveDefinite{N},LogCholeskyMetric},x,y) where N = distance(CholeskySpace{N}(), cholesky(x).L, cholesky(y).L)
+
+@doc doc"""
+    distance(M,x,y)
+
+computes the distance on the [`SymmetricPositiveDefinite`](@ref) manifold between
+`x` and `y` as a [`MetricManifold`](@ref) with [`LogEuclideanMetric`](@ref).
+The formula reads
+
+```math
+    d_{\mathcal P(n)}(x,y) = \lVert \Log x - \Log y \rVert_{\mathrm{F}}
+```
+where $\operatorname{Log}$ denotes the matrix logarithm and $\lVert\cdot\rVert_{\mathrm{F}}$ denotes the
+matrix Frobenius norm.
+"""
+function distance(M::MetricManifold{SymmetricPositiveDefinite{N},LogEuclideanMetric},x,y) where N
+    eX = eigen(Symmetric(x))
+    UX = eX.vectors
+    SX = eX.values
+    eY = eigen(Symmetric(y))
+    UY = eY.vectors
+    SY = eY.values
+    return norm( UX*Diagonal(log.(SX))*transpose(UX) - UY*Diagonal(log.(SY))*transpose(UY))
```

Comment:
Thanks for the careful checks – what do you mean with "this" – all these lines? because from you `@which` recommendation I get one line that I do not understand anything from...

Reply:
Ah, I got it working, I just don't know why it does.

Reply:
Oh, you click on the link that it returns, then you'll be taken to the code in Julia's base that is performed. There, you'll see that basically exactly your code is performed "behind the scenes", so you don't need to do it manually. Basically, there are specialized methods of `log`, `exp` etc. for matrix arguments, and these are _not_ these functions applied elementwise.

Reply:
Ok, I will check somewhen – for now (Julia 1.2, VS Code on Mac OS) I get a file linked and a line of code, but it's not clickable and I am too lazy to look that up in the tree of Julia myself. But I am sure there are better versions and as of now your remarks works just fine. :)